crates/viewer/re_viewer_context/src/callback.rs

```diff
@@ -0,0 +1,120 @@
+use crate::Item;
+use crate::ItemCollection;
+use crate::ItemContext;
+use re_log_types::{TimeReal, Timeline};
+use std::rc::Rc;
+
+// ======================================================================
+// When changing or adding callbacks, grep for the following term:
+//   CALLBACK DEFINITION
+//
+// When changing or adding selection items, grep for the following term:
+//   SELECTION ITEM DEFINITION
+// ======================================================================
+
+// SELECTION ITEM DEFINITION
+/// A single item in a selection.
+#[derive(Debug, serde::Serialize)]
+#[serde(tag = "type")]
+#[serde(rename_all = "snake_case")]
+pub enum SelectionItem {
```

Comment:
The name makes me a bit unhappy - newcomer to this corner of the codebase will 100% be confused about the relationship between `Item`, which is used for selections, and `SelectionItem` which is used for reporting selections in callbacks :spiral_eyes:

Not entirely obvious how to avoid this _elegantly_. How about not elegant:

```suggestion
/// A single item in a selection as reported in callbacks.
#[derive(Debug, serde::Serialize)]
#[serde(tag = "type")]
#[serde(rename_all = "snake_case")]
pub enum CallbackSelectionItem {
```

I figure for the Rust API (i.e. the `rerun` user facing crate) we could still expose it as `pub type SelectionItem = stuff::CallbackSelectionItem` :thinking: